### ISSUE TITLE
API for static HTML export

### ISSUE THREAD

**Issue by jlstevens** [ioam/holoviews#282]
Philipp's has recently pulled out some more functionality out of the ipython subpackage involved in  HTML generation. As this functionality is useful and something we want to support in general (so HoloViews is not necessarily tied to IPython/Jupyter) we should consider implementing a useful API for our users. Here is what I suggest:

```python
from holoviews.plotting import Renderer
data =  Renderer.static_html(obj)  # Dictionary  e.g data['css'], data['js']
html =  Renderer.static_html(obj, html_only=True)  # Standalone HTML
```

One other suggestion is a ``template`` argument which defaults to ``'default'`` (or ``'basic'``?) which defines the overall HTML skeleton. This would allow you to specify a custom template as a formatter with the appropriate fields (i.e ``'css'`` and ``'js'`` in the current example).

Lastly, I might consider splitting the HTML up a little bit more. Perhaps a ``'header'`` field could correspond to the HTML header used to load the appropriate JavaScript libraries?

**Fix:** I forgot to add the ``obj`` argument to define what you want to export!

**Comment by philippjfr** [comment]
For reference [here](http://philippjfr.com/work/work-in-progress/html-export/) is an example of the current API for exporting a standalone HTML file.

**Comment by jlstevens** [comment]
Note that above the methods are on ``Renderer`` (i.e the class) but this should also work for renderer instances. In other words, this will have to be implemented as a ``@bothmethod``

**Comment by nataliabielczyk** [comment]
hi! I am on the user end, and I would like just to ask if this API is still under construction. I tried to run the commands given in the first post of this thread, and I got an error
'type object 'Renderer' has no attribute 'static_html''

**Comment by philippjfr** [comment]
Yes, this has not been implemented yet. Hopefully we'll get around to it soon.

**Comment by philippjfr** [comment]
This has been implemented as part of #311, closing.
